tests/suites/test_suite_psa_crypto.function

```diff
@@ -3152,6 +3152,862 @@ exit:
+
+                part_offset += part_length;
+            }
+        }
+        else
+        {
+            PSA_ASSERT( psa_aead_update_ad(&operation, additional_data->x,
+                                           additional_data->len) );
+        }
+
+        if( data_part_len != -1 )
+        {
+            /* Pass data in parts */
+            part_data_size = PSA_AEAD_UPDATE_OUTPUT_SIZE( key_type, alg,
+                                                     ( size_t ) data_part_len );
+
+            ASSERT_ALLOC( part_data, part_data_size );
+
+            part_offset = 0;
+
+            while( part_offset <= ( input_data->len - tag_length ) )
+            {
+                if( ( input_data->len - tag_length - part_offset ) <
+                    ( uint32_t ) data_part_len )
+                {
+                    part_length =
+                        ( input_data->len - tag_length - part_offset );
+                }
+                else
+                {
+                    part_length = data_part_len;
+                }
+
+                PSA_ASSERT( psa_aead_update( &operation,
+                                             ( input_data->x + part_offset ),
+                                             part_length, part_data,
+                                             part_data_size,
+                                             &output_part_length ) );
+
+                if( output_data2 && output_part_length )
+                {
+                    memcpy( ( output_data2 + part_offset ),
+                            part_data, output_part_length );
+                }
+
+                part_offset += part_length;
+                output_length2 += output_part_length;
+            }
+        }
+        else
+        {
+            PSA_ASSERT( psa_aead_update( &operation, output_data,
+                                             output_length, output_data2,
+                                             output_size2, &output_length2 ) );
+        }
+
+        PSA_ASSERT( psa_aead_verify( &operation, final_data,
+                                     PSA_AEAD_FINISH_OUTPUT_MAX_SIZE,
+                                     &output_part_length,
+                                     tag_buffer, tag_length ) );
+
+        if( output_data2 && output_part_length )
+        {
+            memcpy( ( output_data2 + output_length2 ), final_data,
+                    output_part_length);
+        }
+
+        output_length2 += output_part_length;
+
+        ASSERT_COMPARE( input_data->x, input_data->len,
+                        output_data2, output_length2 );
+    }
+
+exit:
+    psa_destroy_key( key );
+    psa_aead_abort( &operation );
+    mbedtls_free( output_data );
+    mbedtls_free( output_data2 );
+    mbedtls_free( part_data );
+    mbedtls_free( final_data );
+    PSA_DONE( );
+}
+/* END_CASE */
+
+/* BEGIN_CASE */
+void aead_multipart_decrypt( int key_type_arg, data_t *key_data,
```

Comment:
Note: I haven't reviewed `aead_multipart_decrypt` or `aead_multipart_encrypt_decrypt`, I'll do that when my comments in `aead_multipart_encrypt` have been addressed.